src/zenml/zen_server/routers/auth_endpoints.py

```diff
@@ -512,11 +512,12 @@ def api_token(
         resource_type=ResourceType.PIPELINE_RUN, action=Action.CREATE
     )
 
-    if not token.device_id:
-        # If not authenticated with a device, the current API token is returned
-        # as is, without any modifications. Issuing workload tokens is only
-        # supported for device authenticated users, because device tokens can
-        # be revoked at any time.
+    if not token.device_id and not token.api_key_id:
```

Comment:
TODO (@stefannica ): this might break the existing use of pipelines with service accounts.